include/NDDoubleBufferAccessorDecorator.h

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#pragma once
+
+#include "NDRegisterAccessor.h"
+#include "NDRegisterAccessorDecorator.h"
+#include "NumericAddressedBackend.h"
+#include "RegisterInfo.h"
+#include "TransferElement.h"
+
+#include <string>
+
+namespace ChimeraTK {
+
+  template<typename UserType>
+  class NumericDoubleBufferAccessorDecorator : public NDRegisterAccessorDecorator<UserType> {
+   public:
+    using ChimeraTK::NDRegisterAccessorDecorator<UserType>::buffer_2D;
+    using ChimeraTK::NDRegisterAccessorDecorator<UserType>::_target;
+    NumericDoubleBufferAccessorDecorator(const boost::shared_ptr<ChimeraTK::NDRegisterAccessor<UserType>>& target,
+        std::optional<NumericAddressedRegisterInfo::DoubleBufferInfo> doubleBufferConfig,
+        const boost::shared_ptr<DeviceBackend>& backend,
+        std::shared_ptr<NumericAddressedBackend::DoubleBufferControlState> controlState);
```

Comment:
I would move the DoubleBufferControlState into the DoubleBufferAccessorDecorator file. Like that it can also be used by the lmap backend, which should not depend on the NumericAddressedBackend.